rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -291,9 +298,9 @@ void RecorderImpl::record()
 
   serialization_format_ = record_options_.rmw_serialization_format;
   RCLCPP_INFO(node->get_logger(), "Listening for topics...");
-  subscribe_topics(get_requested_or_available_topics());
-
-  if (!record_options_.is_discovery_disabled) {
+  if (record_options_.is_discovery_disabled) {
+    subscribe_topics(get_requested_or_available_topics());
+  } else {
```

Comment:
Nitpick. I would prefer to keep the original logic and add 
```
if (!record_options_.use_sim_time) {
  subscribe_topics(get_requested_or_available_topics());
}
```
For the sake of the case, if someone will decide to simplify or rewrite a logic and for the sake to have a more explicit logic it will be explicitly visible that we need to handle `if (!record_options_.use_sim_time)` as a special case.
In the current implementation the logic about `if (!record_options_.use_sim_time)` is hidden in the upper-level check and via throwing exception.